pandas/tslib.pyx

```diff
@@ -2292,6 +2292,13 @@ cpdef array_to_datetime(ndarray[object] values, errors='raise',
                         iresult[i] = cast_from_unit(val, 'ns')
                     except:
                         iresult[i] = NPY_NAT
+            elif is_bool_object(val) and not is_coerce:
+                if val == NPY_NAT:
+                    iresult[i] = NPY_NAT
+                elif is_raise:
+                    raise TypeError("boolean is not convertible to datetime")
+                else:
+                    return values
```

Comment:
this is wrong

you cannot compare a bool to NPY_NAT (by definition it would never be a bool)
if it hits this at ALL, you have to raise (so that the object path is taken), so I think this can reduce to:

```
elif is_bool_object(val):
    if is_coerce:
         iresult[i] = NPY_NAT
   else:
        raise TypeError("boolean is not convertible to datetime")
```

your tests are not catching this properly